sdk/src/main/java/io/dapr/actors/DaprHttpAsyncClient.java

```diff
@@ -0,0 +1,205 @@
+        String url = String.format(Constants.ACTOR_REMINDER_RELATIVE_URL_FORMAT, actorType, actorId, reminderName);
+        return invokeAPIVoid("PUT", url, data);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Mono<Void> unregisterReminder(String actorType, String actorId, String reminderName) {
+        String url = String.format(Constants.ACTOR_REMINDER_RELATIVE_URL_FORMAT, actorType, actorId, reminderName);
+        return invokeAPIVoid("DELETE", url, null);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Mono<Void> registerTimer(String actorType, String actorId, String timerName, String data) {
+        String url = String.format(Constants.ACTOR_TIMER_RELATIVE_URL_FORMAT, actorType, actorId, timerName);
+        return invokeAPIVoid("PUT", url, data);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Mono<Void> unregisterTimerAsync(String actorType, String actorId, String timerName) {
+        String url = String.format(Constants.ACTOR_TIMER_RELATIVE_URL_FORMAT, actorType, actorId, timerName);
+        return invokeAPIVoid("DELETE", url, null);
+    }
+
+    /**
+     * Invokes an API asynchronously that returns Void.
+     * @param method HTTP method.
+     * @param urlString url as String.
+     * @param json JSON payload or null.
+     * @return Asynchronous Void
+     */
+    private final Mono<Void> invokeAPIVoid(String method, String urlString, String json) {
+        return this.invokeAPI(method, urlString, json).then();
+    }
+
+    /**
+     * Invokes an API asynchronously that returns a text payload.
+     * @param method HTTP method.
+     * @param urlString url as String.
+     * @param json JSON payload or null.
+     * @return Asynchronous text
+     */
+    private final Mono<String> invokeAPI(String method, String urlString, String json) {
+        return Mono.fromSupplier(() -> {
+            try {
+                return tryInvokeAPI(method, urlString, json);
+            } catch (IOException e) {
+                e.printStackTrace();
+                throw new RuntimeException(e);
+            }
+        });
+    }
+
+    /**
+     * Invokes an API synchronously and returns a text payload.
+     * @param method HTTP method.
+     * @param urlString url as String.
+     * @param json JSON payload or null.
+     * @return text
+     */
+    private final String tryInvokeAPI(String method, String urlString, String json) throws IOException {
+        String requestId = UUID.randomUUID().toString();
+        RequestBody body = json != null ? RequestBody.create(MEDIA_TYPE_APPLICATION_JSON, json) : REQUEST_BODY_EMPTY_JSON;
+
+        Request request = new Request.Builder()
+                .url(new URL(this.baseUrl + urlString))
+                .method(method, body)
+                .addHeader(Constants.HEADER_DAPR_REQUEST_ID, requestId)
+                .build();
+
+        Response response = this.httpClient.newCall(request).execute();

```

Comment:
Is this a blocking call?

Reply:
Yes. I have not found a way to make it async from the http client itself. We might end up using RestClient from Microsoft's jar. I would love to fix it but prefer to keep moving on this until we find the solution here.

Reply:
That's fine, we can revisit this line later.

Reply:
Could you please add a TODO comment here that this needs to be async before release?

Reply:
Adding in my next PR.